### PR TITLE
Add test suit

### PR DESCRIPTION
**Pull request by JacopoBonta**
- Add tests for Calculator class

### CODE REVIEW

File: test/Calculator.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import { Calculator, Operation } from '../src/Calculator'
+import { expect } from 'chai'
+import { callbackify } from 'util'
+
+describe('Level 2 - Unit', () => {
+    describe('Calculator', () => {
```
Comment (marmos91) [inline test/Calculator.test.ts]: 🙈Style

File: test/Calculator.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import { Calculator, Operation } from '../src/Calculator'
+import { expect } from 'chai'
+import { callbackify } from 'util'
```
Comment (marmos91) [inline test/Calculator.test.ts]: We don't put spaces in imports

```ts
import {Calculator} from 'src'
```

File: test/Calculator.test.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+describe('Level 2 - Unit', () => {
+    describe('Calculator', () => {
+        
+        let calculator: Calculator
+        let rndOperand: number
```
Comment (marmos91) [inline test/Calculator.test.ts]: 🙈We use `snake_case` for variable declaration. No camelCase

File: test/Calculator.test.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+        beforeEach(() => {
+            calculator = new Calculator()
+            rndOperand = Math.floor(Math.random() * 100 - 50)
```
Comment (marmos91) [inline test/Calculator.test.ts]: semicolons

Comment (marmos91) [reply test/Calculator.test.ts]:
```py
[print 'semicolons!' for line in file]
```

File: test/Calculator.test.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import {Calculator, Operation} from '../src/Calculator';
+import {expect} from 'chai';
+
+describe('Level 2 - Unit', () =>
+{
+    describe('Calculator', () =>
+    {
+
```
Comment (marmos91) [inline test/Calculator.test.ts]: 🙈Useless blank line

**Comment by JacopoBonta** [comment]
- added Webpage unit tests

File: test/Webpage.test.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                    path: '/test',
+                    reply: {
+                        status: 200,
+                        headers: { 'content-type': 'text/html'},
+                        body: '<h1>Hello, Cubbit!</h1>'
+                    }
+                });
+
+                webpage.getWebpage('http://localhost:9000/test')
+                    .then((body) =>
```
Comment (marmos91) [inline test/Webpage.test.ts]: 🙈Putting this on the line above would save us a level of indentation

File: test/Webpage.test.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+            afterEach(() =>
+            {
+                get_webpage.restore();
+                write_file.restore();
+            });
+
+            it('should call getWebpage and _writeFile once', (done) =>
+            {
+                webpage.saveWebpage(url, path)
+                    .then(() =>
```
Comment (marmos91) [inline test/Webpage.test.ts]: Putting this on the line above would save us a level of indentation

File: test/Webpage.test.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+            after((done) => server.stop(done));
+        });
+
+        describe('saveWebpage method', () =>
+        {
+            let get_webpage, write_file, webpage
+                , url = 'http://localhost:9000/foo'
+                , path = 'my/cool/path'
+                , content = 'some content';
```
Comment (marmos91) [inline test/Webpage.test.ts]: 🙈put commas before new lines

**Pull request closed without merging**